_posts/2024-02-18-how-its-made.md

```diff
@@ -0,0 +1,180 @@
+As the source code of the game is freely accessible, this can be literally anyone, for a multitude of reasons.
+
+Many features come about because someone has an idea how their game could be better and decides to work on it.
+Bugs are usually fixed because someone either feels responsible for it in some way or is just plain annoyed by it.
+But really, everybody has a different personal motivation.
+There's one reason you're not going to find though: Because the boss said so, or because it is on the big company road map.
+OpenTTD is purely developed by volunteers, there's no company, foundation, or other controlling central entity behind it.
+
+Now let's assume that the person is in fact developing a feature (like for example the new ship pathfinder) and has completed the initial development.
+Since OpenTTD moved to its current home on GitHub almost 6 years ago, we've adopted a fixed process for any change to OpenTTD.
+The first step in the process is to open what is called a *pull request* (PR) on GitHub with the proposed changes to the source code in the form of git commits.
+Additionally, we require a description of the changes following a specific template to make sure nothing important is left out.
+
+Whenever a PR is opened, various automatic checks run to verify some formal things, but while those checks are running, other contributors can already begin to review and comment on the pull request.
+The automated checks include what is called *continuous integration* (CI), which ensures the PR actually compiles on all platforms we support and not just the one the author is working on.
+
+![Pull request overview]({% link /static/img/post_2024-02-18-how-its-made/PR.png %}){:width="100%"}
+
+Automated checks are well and good, but so far they can't replace a human code review.
+In code review, other people look at the code and try to find any issues or spots that could be improved.
+This can be everything from validating functionality, checking UI and user experience, to proof-reading code comments.
+For a large change like the ship pathfinder, we've even organized a multi-player test game to check for any dreaded desync problems.
+
+Usually, this results in a back and forth between the author updating the PR and the reviewers looking again until the PR is either declared good or rejected.
+Unfortunately, not every author has the motivation to follow this until the end, so it happens that PRs just kind of fizzle off without reaching a finished state.
+Up to here, everybody can pitch in with reviewing a PR, but actually approving or rejecting a PR requires a decision by someone that has commit access to the OpenTTD repository.
+
+PR aren't actually outright rejected that often.
+Reasons for rejection might be that the PR is contrary to the goals of the OpenTTD project, like e.g. re-creating the original Transport Tycoon gameplay.
```

Comment:
This is phrased oddly. To me, the example should be a reason the PR is rejected, not an example of one of the project goals.

```suggestion
Reasons for rejection might be that the PR is contrary to the goals of the OpenTTD project, like trying to add new content that should be done as an add-on NewGRF.
```